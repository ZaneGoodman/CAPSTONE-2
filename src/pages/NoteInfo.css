.NoteInfo-note-header {
  text-align: center;
  align-items: center;
}
