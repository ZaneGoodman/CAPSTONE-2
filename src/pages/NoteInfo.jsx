import React, { useState } from "react";
import Note from "./Note";
import Accordion from "react-bootstrap/Accordion";
import "./NoteInfo.css";

//Show Note Info, toggle note on click
const NoteInfo = ({ idx, date, season, has_prayed, notes }) => {
  return (
    <Accordion defaultActiveKey={["0"]} alwaysOpen>
      <Accordion.Item eventKey={idx}>
        <Accordion.Header title="noteInfo">
          {date.slice(0, 10)} | {season} {has_prayed ? "🙏🏽📿" : "👎🏽"}
        </Accordion.Header>
        <Accordion.Body>
          <Note note={notes} />
        </Accordion.Body>
      </Accordion.Item>
    </Accordion>
  );
};
export default NoteInfo;
