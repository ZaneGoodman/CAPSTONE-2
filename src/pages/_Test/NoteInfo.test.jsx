import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import AuthProvider from "../../provider/authProvider";
import NoteInfo from "../NoteInfo";
//Smoke test
test("renders NoteInfo component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <NoteInfo
          date="12/05/05"
          season="Ordinary TIme"
          has_prayed={true}
          notes="Test Note"
        />
      </MemoryRouter>
    </AuthProvider>
  );
});

//Snapshot test  <MemoryRouter>
it("NoteInfo matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <NoteInfo
          date="12/05/05"
          season="Ordinary TIme"
          has_prayed={true}
          notes="Test Note"
        />
      </MemoryRouter>
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});

test("renders Note component on click of span", async () => {
  const { getByTitle, getByText } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <NoteInfo
          date="12/05/05"
          season="Ordinary TIme"
          has_prayed={true}
          notes="Test Note"
        />
      </MemoryRouter>
    </AuthProvider>
  );

  const accordianHeader = getByTitle("noteInfo");
  fireEvent.click(accordianHeader);
  expect(getByText("Test Note")).toBeInTheDocument();
});
