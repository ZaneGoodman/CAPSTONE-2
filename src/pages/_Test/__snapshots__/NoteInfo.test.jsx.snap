// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteInfo matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="accordion"
  >
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
        title="noteInfo"
      >
        <button
          aria-expanded="false"
          class="accordion-button collapsed"
          type="button"
        >
          12/05/05 | Ordinary TIme 🙏🏽📿
        </button>
      </h2>
      <div
        class="accordion-collapse collapse"
      >
        <div
          class="accordion-body"
        >
          <p>
            Test Note
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
